src/platform/crypt_openssl.c

```diff
@@ -34,13 +34,75 @@
 #include "openssl/rsa.h"
 #include "openssl/ssl.h"
 #include "openssl/x509.h"
+#include "openssl/evp.h"
+
 #ifdef _WIN32
 #pragma warning(pop)
 #endif
 #ifdef QUIC_CLOG
 #include "crypt_openssl.c.clog.h"
 #endif
 
+EVP_CIPHER *CXPLAT_AES_128_GCM_ALG_HANDLE;
+EVP_CIPHER *CXPLAT_AES_256_GCM_ALG_HANDLE;
+EVP_CIPHER *CXPLAT_AES_128_ECB_ALG_HANDLE;
+EVP_CIPHER *CXPLAT_AES_256_ECB_ALG_HANDLE;
+EVP_CIPHER *CXPLAT_CHACHA20_POLY1305_ALG_HANDLE;
+#ifdef IS_OPENSSL_3
+EVP_MAC_CTX *CXPLAT_HMAC_SHA256_CTX_HANDLE;
+EVP_MAC_CTX *CXPLAT_HMAC_SHA384_CTX_HANDLE;
+EVP_MAC_CTX *CXPLAT_HMAC_SHA512_CTX_HANDLE;
+
```

Comment:
Not sure why, but macOS static linking build is complaining about a bunch of duplicate symbols:
```
/Applications/Xcode_12.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Og -gdwarf -arch x86_64 -isysroot /Applications/Xcode_12.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -mmacosx-version-min=10.15 -Wl,-search_paths_first -Wl,-headerpad_max_install_names CMakeFiles/quicipserver.dir/quicipserver.cpp.o -o ../../../../../../../artifacts/bin/macos/x64_Debug_openssl_Static/quicipserver  ../../../../../../../artifacts/bin/macos/x64_Debug_openssl_Static/libmsquic.a -framework CoreFoundation -framework Security 
duplicate symbol '_ENGINE_new' in:
    ../../../../../../artifacts/bin/macos/x64_Debug_openssl_Static/libmsquic.a(eng_lib.o)
    ../../../_deps/opensslquic-build/openssl/lib/libcrypto.a(eng_lib.o)
duplicate symbol '_engine_cleanup_add_first' in:
    ../../../../../../artifacts/bin/macos/x64_Debug_openssl_Static/libmsquic.a(eng_lib.o)
    ../../../_deps/opensslquic-build/openssl/lib/libcrypto.a(eng_lib.o)
duplicate symbol '_engine_cleanup_add_last' in:
    ../../../../../../artifacts/bin/macos/x64_Debug_openssl_Static/libmsquic.a(eng_lib.o)
    ../../../_deps/opensslquic-build/openssl/lib/libcrypto.a(eng_lib.o)
duplicate symbol '_engine_cleanup_int' in:
    ../../../../../../artifacts/bin/macos/x64_Debug_openssl_Static/libmsquic.a(eng_lib.o)
    ../../../_deps/opensslquic-build/openssl/lib/libcrypto.a(eng_lib.o)
duplicate symbol '_engine_lock_init' in:
    ../../../../../../artifacts/bin/macos/x64_Debug_openssl_Static/libmsquic.a(eng_lib.o)
    ../../../_deps/opensslquic-build/openssl/lib/libcrypto.a(eng_lib.o)
duplicate symbol '_ENGINE_set_cmd_defns' in:
    ../../../../../../artifacts/bin/macos/x64_Debug_openssl_Static/libmsquic.a(eng_lib.o)
    ../../../_deps/opensslquic-build/openssl/lib/libcrypto.a(eng_lib.o)
...
```

Reply:
This is an error on our build scripts. In src/tools/pcp/CMakeLists.txt, chane the target_link_libraries line to 

```
if (BUILD_SHARED_LIBS)
target_link_libraries(quicpcp core platform)
endif()
```